Mozilla Hubs case (#1529)
elixir-lang/elixir-lang.github.com: _posts/2021-06-02-social-virtual-spaces-with-elixir-at-mozilla.markdown

---
layout: post
title: Social virtual spaces with Elixir at Mozilla
author: José Valim
category: Elixir in Production
excerpt: A case study of how Elixir is being used at Mozilla.
logo: /images/cases/logos/mozilla-hubs.png
tags: virtual-spaces phoenix
---

> *Welcome to our series of case studies about companies using Elixir in production. [See all cases](/cases.html) we have published so far.*

Reviewer: I believe this is the correct form, but not 100% sure 😅 

```suggestion
*Welcome to our series of case studies about companies using Elixir in production. [See all the cases](/cases.html) we have published so far.*
```

Author: I think we don't need the article as we are not specifying any cases in particular? 🤔

[Hubs](https://hubs.mozilla.com/) is Mozilla's take on virtual social experiences. You build your own private spaces and share them with your friends, co-workers, and community. Avatars in this space can move freely in a 3D social environment and watch videos, exchange messages, and talk to other people nearby. All you need is a browser and a microphone!

> Hubs is [fully open source](https://github.com/mozilla/hubs) and you can host it on your infrastructure via [Hubs Cloud](https://hubs.mozilla.com/cloud). Community managers, educators, and event organizers have been using Hubs Cloud to run virtual events and online activities tailored to their specific brandings and needs. All it takes to run your own version of Hubs is one click away - which perhaps makes Hubs the most deployed Phoenix application ever!

Reviewer 2: Link to the elixir repo? https://github.com/mozilla/reticulum

> Hubs is [fully open source](https://github.com/mozilla/hubs) and you can host it on your infrastructure via [Hubs Cloud](https://hubs.mozilla.com/cloud). Community managers, educators, and event organizers have been using Hubs Cloud to run virtual events and online activities tailored to their specific brandings and needs. All it takes to run your own version of Hubs is one click away - which perhaps makes Hubs the most deployed Phoenix application ever!

Author: ```suggestion
Hubs is [fully](https://github.com/mozilla/hubs) [open source](https://github.com/mozilla/reticulum) and you can host it on your infrastructure via [Hubs Cloud](https://hubs.mozilla.com/cloud). Community managers, educators, and event organizers have been using Hubs Cloud to run virtual events and online activities tailored to their specific brandings and needs. All it takes to run your own version of Hubs is one click away - which perhaps makes Hubs the most deployed Phoenix application ever!
```

> Hubs is [fully open source](https://github.com/mozilla/hubs). You can also host it on your infrastructure via [Hubs Cloud](https://hubs.mozilla.com/cloud). Community managers, educators, and event organizers have been using Hubs Cloud to run virtual events and online activities tailored to their specific brandings and needs. All it takes to run your own version of Hubs is a button click - which perhaps makes Hubs the most deployed Phoenix application ever!

Reviewer: ```suggestion
Hubs is [fully open source](https://github.com/mozilla/hubs). You can also host it on your infrastructure via [Hubs Cloud](https://hubs.mozilla.com/cloud). Community managers, educators, and event organizers have been using Hubs Cloud to run virtual events and online activities tailored to their specific brandings and needs. All it takes to run your own version of Hubs is one click away - which perhaps makes Hubs the most deployed Phoenix application ever!
```

![Mozilla Hubs](/images/cases/bg/mozilla-hubs.jpg)

## From VR to Elixir

The Hubs team started at Mozilla as the Mixed Reality team about 3.5 years ago. Their main goal was to explore ways for online social interaction via avatars and mixed reality.

> They quickly focused on building their first proof of concept, where avatars could communicate, move around, and join different rooms, everything running directly from the browser. This was a significant departure from the state of the art of Virtual Reality everywhere, as the getting started experience up to this point was cumbersome and often required steep investment in the form of VR headsets.

Reviewer: ```suggestion
They quickly focused on building their first proof of concept, where avatars could communicate, move around, and join different rooms, everything running directly in the browser. This was a significant departure from the state of the art of Virtual Reality everywhere, as the getting started experience up to this point was cumbersome and often required steep investment in the form of VR headsets.
```
